### PR TITLE
Add `ripgrep-13.0.0` as a new benchmark

### PR DESCRIPTION
**Pull request by Kobzol**
It's slower to compile than the old `ripgrep`, although it seems to me that most of the time was spent by compiling dependencies (`ripgrep` has only 5k lines of code).
```
ripgrep check 6.8
ripgrep debug 10.07
ripgrep release 20.97

ripgrep-13.0.0 check 12.38
ripgrep-13.0.0 debug 18.08
ripgrep-13.0.0 release 25.31
```

### CODE REVIEW

File: collector/benchmarks/README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ They mostly consist of real-world crates.
   compiler in certain ways.
 - **piston-image**: A modular game engine. An interesting Rust program.
 - **regex**: A regular expression parser. Used by many Rust programs.
-- **ripgrep**: A line-oriented search tool. A widely-used utility.
+- **ripgrep-13.0.0**: A line-oriented search tool. A widely-used utility.
```
Comment (nnethercote) [inline collector/benchmarks/README.md:38]: Strictly speaking, you should leave the old `ripgrep` there until it's removed in a later PR.

File: collector/benchmarks/ripgrep/Cargo.lock [hunk 2]
```diff
@@ -1,417 +1,419 @@
+# This file is automatically @generated by Cargo.
+# It is not intended for manual editing.
+version = 3
```
Comment (nnethercote) [inline collector/benchmarks/ripgrep/Cargo.lock:3]: This file shouldn't be changed.

**Review by nnethercote — CHANGES_REQUESTED** [review]
Thank you for doing this. Mostly looks good, just a couple of minor things that need fixing.

**Comment by bjorn3** [comment]
Isn't most of the logic of ripgrep in the grep library crate?

**Comment by Kobzol** [comment]
It is, I suppose that ripgrep is really mostly a thin wrapper over `grep` + `regex`.

**Comment by nnethercote** [comment]
I compared `ripgrep` and `ripgrep-13.0.0`, just the time for the final crate:

Benchmark & Profile | Scenario | % Change | Significance Factor? | [object HTMLSpanElement] | [object HTMLSpanElement]
-- | -- | -- | -- | -- | --
helloworld check | full | 0.00% | - | 0.16 | 0.16
helloworld opt | full | 0.00% | - | 0.22 | 0.22
helloworld debug | full | 0.00% | - | 0.23 | 0.23
ripgrep check | full | 0.00% | - | 0.53 | 0.53
ripgrep debug | full | 0.00% | - | 1.38 | 1.38
ripgrep opt | full | 0.00% | - | 3.67 | 3.67
ripgrep-13.0.0 check | full | 0.00% | - | 0.48 | 0.48
ripgrep-13.0.0 debug | full | 0.00% | - | 1.75 | 1.75
ripgrep-13.0.0 opt | full | 0.00% | - | 4.13 | 4.13

It's not that different. So I think it's still a reasonable benchmark to have.

@Kobzol I had to add a `"touch_file": "crates/core/main.rs"` to `ripgrep-13.0.0/perf-config.json` to get the `bench_local` run to work. Have you run the new benchmark under `collector`?

File: collector/benchmarks/ripgrep/Cargo.lock (hunk 2 above)
Comment (Kobzol) [reply collector/benchmarks/ripgrep/Cargo.lock:3]: Oops, good catch, thanks.

**Comment by Kobzol** [comment]
@nnethercote I ran it under `bench_local` and it did work. The `perf-config.json` should already contain `"touch_file": "crates/core/main.rs"` :thinking:

**Review by nnethercote — APPROVED** [review]
The changes are good, though I'm worried about the weirdness with the commit message not looking normal.

**Pull request merged**